streamlit_lottie/__init__.py

```diff
@@ -17,75 +26,155 @@
     _st_lottie = components.declare_component("streamlit_lottie", path=build_dir)
 
 
-def st_lottie(
-    animation_data,
-    speed=1,
-    reverse=False,
-    loop=True,
-    quality="low",
-    height=None,
-    width=None,
-    key=None,
-):
-    """Create a new instance of "my_component".
+class LottieDownloadFailure(StreamlitAPIException):
+    pass
+
+
+def _download_animation_data(url):
+    request = requests.get(url)
+    try:
+        return request.json()
+    except (json.JSONDecodeError, TypeError) as exc:
+        raise LottieDownloadFailure(
+            f"""Unable to download animation data from {url}  \n
+* status code {request.status_code}
+* JSONDecodeError {exc}
+"""
+        )
+
+
+def download_animation_data(url):
+    try:
+        return _download_animation_data(url)
+    except LottieDownloadFailure:
+        time.sleep(1)
+        return _download_animation_data(url)
+
+
+def get_animation_data(animation_source: Union[bytes, str, dict]):
+    if not (
+        isinstance(animation_source, bytes)
+        | isinstance(animation_source, str)
+        | isinstance(animation_source, dict)
+    ):
+        raise StreamlitAPIException(
+            f"""Animation data must be one of Lottie URL or loaded JSON represented by dict or string/bytes UTF-8 JSON representative.  \n
+    Given type is: {type(animation_source)}"""
+        )
+
+    if isinstance(animation_source, bytes):
+        animation_source = animation_source.decode("UTF-8")
+
+    animation_data = None
+    if isinstance(animation_source, dict):
+        animation_data = animation_source
+    elif isinstance(animation_source, str):
+        try:
+            if validate_url(animation_source):
+                animation_data = download_animation_data(animation_source)
+        except ValidationFailure:
+            # Is not url try to covert it to json
```

Comment:
covert --> convert

Reply:
It's fixed 👍 Thank you for review and noticing 🙂